codec-http/src/main/java/io/netty/handler/codec/http/ClientCookieEncoder.java

```diff
@@ -119,6 +119,15 @@ private static void encode(StringBuilder buf, Cookie c) {
         // rawValue > value > ""
         String value = c.rawValue() != null ? c.rawValue()
                 : c.value() != null ? c.value() : "";
+
+        if (ServerCookieDecoder.containsInvalidCookieNameOctet(c.name(), 0, c.name().length())) {
```

Comment:
It looks very strange to call ServerCookieDecoder from the ClientEncoder. Maybe introduce CookieUtil?